lib/file_database.rb

```diff
@@ -18,21 +18,40 @@ def database_empty?
 
   def create(contact)
     file_array = all.push(contact)
-
-    file.rewind
-    file.truncate(0)
-    file << file_array.to_json
-    file.flush
+    write_to_file(file_array)
   end
 
   def count
     all.length
   end
 
+  def contact_at(index)
+    all[index]
+  end
+
+  def update(index, new_data)
+    file_array = all
+
+    file_array[index].update(new_data)
+    write_to_file(file_array)
+  end
+
+  def delete(index)
+    file_array = all.delete_at(index)
```

Comment:
doesn't `delete_at` return the element that was deleted, rather than the array without the element :thinking:

Reply:
ha, my test was that bad it didn't even see that